lib/puppet/type/concat_file.rb

```diff
@@ -134,6 +140,11 @@ def fragment_content(r)
       tmp = Puppet::FileServing::Content.indirection.find(@source, :environment => catalog.environment)
       fragment_content = tmp.content unless tmp.nil?
     end
+
+    if self[:ensure_newline]
+      fragment_content<<"\n"
```

Comment:
I think we should be checking to see if it has a newline before appending a newline.  If it already ends in \n we will end up making a double newline.